File: EEProjects/ShopWebApp/ui-shop-app/src/app/core/services/product.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';

import { ApiService } from './api.service';
import { Product } from '../models/product.model';
import { PageData } from '../models';


@Injectable()
export class ProductService {
  constructor (
    private apiService: ApiService,
    private http: HttpClient,
  ) {}

  getAll(page = 0, size = 10): Observable<PageData> {
    return this.apiService.get(`/products?page=${page}&size=${size}`);
  }

  getItem(itemId: number): Observable<Product> {
    return this.apiService.get(`/products/${itemId}`);
  }

  update(product: any): Observable<Product> {
    return this.apiService.put(`/products`, product);
  }

  add(product: any): Observable<Product> {
    return this.apiService.post(`/products`, product);
  }
}

File: EEProjects/ShopWebApp/ShopWebApp/src/main/java/org/yakushev/shopwebapp/service/ProductService.java
package org.yakushev.shopwebapp.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.yakushev.shopwebapp.model.Product;

public interface ProductService {

	Page<Product> getAll(Pageable pageable);

	Product getById(Long id);

	Product add(Product value);

	Product update(Product value);

}
